### DISCUSSION TITLE
Difficulties getting heightmap import to work

### DISCUSSION THREAD

**Discussion by Mysterius** [TokisanGames/Terrain3D#263] [Questions] [answered]
Hello,

Sorry to probably waste your time, I'm a complete newbie. I'm trying to import a heightmap generated by https://heightmap.skydark.pl/ but the result is weird. It's in .RAW format.

Here is what I get with (what I think are) correct settings :
![Godot_v4 2-rc1_mono_win64_cNxyNycx4h](https://github.com/TokisanGames/Terrain3D/assets/4077266/06ed6f5c-a594-40d6-8dbb-dc0341996d67)

If I lower the scale, I can see that the general shape is good, but the terrain seems to be composed of small dots that have extreme height ranges.
![Godot_v4 2-rc1_mono_win64_hHtJrOKroo](https://github.com/TokisanGames/Terrain3D/assets/4077266/5a89d04c-ce9d-41c2-ab3a-3bea556d8657)

When opened in Photoshop, the heightmap seems normal, in classical shades of grey.

What am I doing wrong ?

**Comment by Mysterius** [comment]
For reference, here's the heightmap I'm using : https://www.file.io/zuha/download/I9gDH7z1dV05

**Comment by TokisanGames** [comment, marked as answer]
We interpret raw as r16, which Krita reads and writes fine and that's the way it comes out of 3rd party tools like world machine. [Photoshop raw](https://helpx.adobe.com/photoshop/using/file-formats.html#photoshop_raw_format) is a different format. I haven't tried it until now. If you try to open it in krita as r16 you'll see essentially the same thing. If you save out of photoshop in non-interleaved mode then open as r16 in krita, you'll see the image properly but in quarters. Basically don't use Photoshop Raw. Use EXR out of photoshop.

Also your heightmap is 8-bit. That's going to look like garbage. Use 16-bit or higher for heights.

**Reply by TokisanGames** [reply]
I've updated the docs in the pending 0.9 to say don't use Photoshop raw and use the free exr plugin https://www.exr-io.com/

**Reply by Mysterius** [reply]
Thank you for your clear explanations and updating the docs, I was able to get it working then :)
